corelog/logmap.go

```diff
@@ -0,0 +1,188 @@
+package corelog
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"sync"
+)
+
+// LogLevel is the set of all log levels.
+type LogLevel int8
+
+const (
+	// CRITICAL is the lowest log level; only errors which will end the program will be propogated.
+	CRITICAL LogLevel = -1
+	// ERROR is for errors that are not fatal but lead to troubling behavior.
+	ERROR = 0
+	// WARNING is for errors which are not fatal and not errors, but are unusual. Often sourced from misconfigurations.
+	WARNING = 1
+	// INFO is a log level for common, everyday log updates.
+	INFO = 2
+	// DEBUG is the default hidden level for more verbose updates about internal processes.
+	DEBUG = 3
+	// VERBOSE is for (potentially) call by call tracing of programs.
+	VERBOSE = 4
+)
+
+// Char returns a single-character representation of the log level.
+func (l LogLevel) Char() string {
+	switch l {
+	case CRITICAL:
+		return "C"
+	case ERROR:
+		return "E"
+	case WARNING:
+		return "W"
+	case INFO:
+		return "I"
+	case DEBUG:
+		return "D"
+	case VERBOSE:
+		return "V"
+	default:
+		panic("Unhandled loglevel")
+	}
+}
+
+// ParseLevel translates some potential loglevel strings into their corresponding levels.
+func ParseLevel(s string) LogLevel {
+	switch s {
+	case "ERROR", "0", "E":
+		return ERROR
+	case "WARNING", "1", "W":
+		return WARNING
+	case "INFO", "2", "I":
+		return INFO
+	case "DEBUG", "3", "D":
+		return DEBUG
+	case "VERBOSE", "4", "V":
+		return VERBOSE
+	}
+	return CRITICAL
```

Comment:
Seems like a value indicating a parse error would be better, otherwise typoing a string will be accepted as valid.